src/Error/ExceptionTrap.php

```diff
@@ -185,6 +185,83 @@ public function handleException(Throwable $exception): void
         }
     }
 
+    /**
+     * Shutdown handler
+     *
+     * Convert fatal errors into exceptions that we can render.
+     *
+     * @return void
+     */
+    public function handleShutdown(): void
+    {
+        $megabytes = $this->_config['extraFatalErrorMemory'] ?? 4;
+        if ($megabytes > 0) {
+            $this->increaseMemoryLimit($megabytes * 1024);
+        }
+        $error = error_get_last();
+        if (!is_array($error)) {
+            return;
+        }
+        $fatals = [
+            E_USER_ERROR,
+            E_ERROR,
+            E_PARSE,
+        ];
+        if (!in_array($error['type'], $fatals, true)) {
+            return;
+        }
+        $this->handleFatalError(
+            $error['type'],
+            $error['message'],
+            $error['file'],
+            $error['line']
+        );
+    }
+
+    /**
+     * Increases the PHP "memory_limit" ini setting by the specified amount
+     * in kilobytes
+     *
+     * @param int $additionalKb Number in kilobytes
+     * @return void
+     */
+    public function increaseMemoryLimit(int $additionalKb): void
+    {
+        $limit = ini_get('memory_limit');
+        if ($limit === false || $limit === '' || $limit === '-1') {
+            return;
+        }
+        $limit = trim($limit);
+        $units = strtoupper(substr($limit, -1));
+        $current = (int)substr($limit, 0, strlen($limit) - 1);
+        if ($units === 'M') {
```

Comment:
To be 100% compatible, this should use all digits as the limit and the compare the remaining string with defaulting to bytes.

https://www.php.net/manual/en/faq.using.php#faq.using.shorthandbytes